examples/lighting-app/silabs/data_model/lighting-thread-app.matter

```diff
@@ -2063,6 +2322,10 @@ endpoint 0 {
   }
 
   server cluster GeneralDiagnostics {
+    emits event HardwareFaultChange;
+    emits event RadioFaultChange;
+    emits event NetworkFaultChange;
```

Comment:
No, not yet. No event is actually generated for those optional fields. I am planning on adding an app code to generate the events. This applies to both thread and wifi versions.

Reply:
Feels like we should enable these when the app code is being added....

Reply:
Some mandatories were missing with the correct value and the power source cluster was missing which has a mandatory event. 

I don't know what the issue is with merging this commit now and using and adding code that generates events later. Event list attributes will be available instead of erroring out. Fault events won't be sent today with this PR but I don't see that as an issue. most of those fault occurs extremely rarely.

Reply:
I think the basic question keeps coming back to what EventList means: does it mean the app _actually emits_ the event as needed?  If so, things that are not emitted should not be in EventList.

Reply:
> If so, things that are not emitted should not be in EventList.
Most events should be disabled in most apps then except some test functions/commands generating events on the all app clusters..

Anyhow. If you insist, I'll push the event emitters in the app code in this PR instead of a separate one.

Reply:
To be clear: I am just saying the spec does not in fact define what EventList contains....

Reply:
I only kept the boot reason event out of the general diagnostic cluster events list. The three others are optional and require more work to get error monitoring in the hardware, radio or network stacks.
We might add those back at a later time